app/javascript/packs/document-capture.jsx

```diff
@@ -11,6 +12,19 @@ function getMetaContent(name) {
   return document.querySelector(`meta[name="${name}"]`)?.content ?? null;
 }
 
+/** @type {import('../app/document-capture/context/device').DeviceContext} */
+const device = {
+  supports: {
+    video: {
+      facingMode: {
+        environment:
+          'mediaDevices' in window.navigator &&
+          /ip(hone|ad|od)|android/i.test(window.navigator.userAgent),
+      },
+    },
+  },
```

Comment:
QQ, do we need all this nesting? It seems like we only really have two true/false values (environment, user), and the `video` and `supports` seem otherwise unused, so what if we just had:
```
device = {
  videoFacingMode: {
    environment: true,
    user: true
  }
}
```

Reply:
> QQ, do we need all this nesting? It seems like we only really have two true/false values (environment, user), and the `video` and `supports` seem otherwise unused, so what if we just had:

I've been going back and forth on this. The original idea was to model it largely after the [MediaStreamConstraints](https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamConstraints) shape where, for example, one might call [`getUserMedia`](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia) as:

```js
getUserMedia( { video: { facingMode: { exact: 'environment' } } } );
```

...and then `supports` as its own key to allow room for some future scenario where we could want to use this context to retrieve general information about the device.

But it may be a bit premature to go down that path for a few reasons. I'm also seeing ([example](https://github.com/18F/identity-idp/pull/3994#discussion_r463036821)) a lot of variability in how we need to consider "mobile", where device support may mean many things:

- "Is mobile", regardless of camera availability
- Camera availability
- Acuant support
- (Future) "Is desktop" with camera availability

For the short-term, then, maybe it's something where we just keep this as a simple boolean flag (`isMobile`), and then refactor it later as those details become clearer.

Reply:
Yeah I'm a big fan of starting simple and adding layers when they're needed

Reply:
Simplified in b90f970.